<launch>
	<arg name="tf_prefix"    default="mur216" />
	<arg name="robot_x"   default="0.0" />
	<arg name="robot_y"   default="0.0" />
	<arg name="robot_yaw" default="0.0" />
	<arg name="model_name" default="mur216"/>
	<arg name="move_base_selector" default="move_base" />
	<arg name="global_planner" default="GlobalPlanner" />
	<arg name="local_planner" default="DWAPlannerROS" />
	<arg name="robot_urdf_path" value="$(find mur_description)/urdf/mur_216.gazebo.xacro"/>
	<arg name="prefix" value="UR16/" doc="Prefix for the UR16 links"/>

	<group ns = "$(arg model_name)">
		<rosparam file="$(find mur_moveit_config)/config/arm_controller.yaml" command="load" subst_value="true"/> 
		<rosparam file="$(find mur_control)/config/mur216e_controllers.yaml" command="load" subst_value="true"/> 
	
	
		<include file="$(find mir_launch_sim)/launch/utils/general_mir.launch">
			<arg name="tf_prefix" value="$(arg tf_prefix)"/>
			<arg name="robot_x"  value="$(arg robot_x)"/>
			<arg name="robot_y"  value="$(arg robot_y)"/>
			<arg name="robot_yaw"  value="$(arg robot_yaw)"/>
			<arg name="model_name" value="$(arg model_name)"/>
			<arg name="robot_urdf_path" value="$(arg robot_urdf_path)"/>
			<!-- <arg name="add_spawn_args" value="-J UR16/shoulder_lift_joint -0.7 -J UR16/elbow_joint 1.5 -J UR16/shoulder_pan_joint 0.0 -J UR16/wrist_1_joint -2.4 -J UR16/wrist_2_joint -1.5621"/> -->
			<!-- 1.5601472624388606, -2.587638077291936, -1.363669475276887, -0.9665857266916063, 1.5712602500272297, -2.1364874278617663e-06 -->
			<arg name="add_spawn_args" value="-J UR16/shoulder_pan_joint 1.5678655285071859 -J UR16/shoulder_lift_joint -2.587638077291936 -J UR16/elbow_joint -1.363669475276887 -J UR16/wrist_1_joint -0.9665857266916063 -J UR16/wrist_2_joint 1.5712602500272297"/>

			<arg name="move_base_selector" value="$(arg move_base_selector)"/>
			<arg name="global_planner" value="$(arg global_planner)" />
			<arg name="local_planner" value="$(arg local_planner)" />
		</include>
	

			<!-- Start all necessary moveit files -->
		<include file="$(find mur_moveit_config)/launch/start_moveit.launch">
			<arg name="model_name" value="$(arg model_name)"/>
			<arg name="prefix" value="$(arg prefix)"/>
		</include>

		<node name="ur_controller_spawner" pkg="controller_manager" type="spawner" output="screen"
			args="joint_group_vel_controller">
		</node>

		<node name="ur_controller_stopped_spawner" pkg="controller_manager" type="spawner" output="screen"
			args="--stopped arm_controller">
		</node>

        <param name="prefix" value="$(arg prefix)"/>
        <node pkg="mur_sb" type="current_pos_pub.py" name="current_pos_pub" output="screen"/>


		<!-- <arg name="node_start_delay" default="20.0" />
		<node pkg="controller_manager" type="spawner" name="spawner_joint_vel_control" args="joint_group_vel_controller" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
		</node> -->
		
	</group> 

</launch>
